test/tests/functional/pbs_svr_dyn_res.py

```diff
@@ -480,3 +480,237 @@ def test_res_size_runtime(self):
+
+        # Script returns float value for resource type 'string'
+        script_body = "echo 1.25"
+
+        fn = self.du.create_temp_file(prefix="PtlPbs_string",
+                                      suffix=".scr",
+                                      body=script_body)
+        self.du.chmod(path=fn, mode=0755, sudo=True)
+
+        resval = ['"' + resname[0] + ' ' + '!' + fn + '"']
+
+        self.setup_dyn_res(resname, restype, resval)
+
+        # Submit job
+        a = {'Resource_List.foo': 'abc'}
+        j = Job(TEST_USER, attrs=a)
+        jid = self.server.submit(j)
+
+        # The job shouldn't run
+        job_comment = "Can Never Run: Insufficient amount of server resource:"
+        job_comment += " foo (abc != 1.25)"
+        a = {'job_state': 'Q', 'comment': job_comment}
+        self.server.expect(JOB, a, id=jid, attrop=PTL_AND)
+
+    def test_res_string_array_invalid_input1(self):
+        """
+        Test invalid values returned from server_dyn_resource
+        script for resource type 'string_array'
+        Script returns a 'string' instead of type 'string_array'.
+        """
+        # Create a resource of type string_array
+        resname = ["foo"]
+        restype = ["string_array"]
+
+        # Prep for server_dyn_resource script
+        script_body = "echo red blue"
+
+        fn = self.du.create_temp_file(prefix="PtlPbs_str_arr",
+                                      suffix=".scr",
+                                      body=script_body)
+        self.du.chmod(path=fn, mode=0755, sudo=True)
+
+        resval = ['"' + resname[0] + ' ' + '!' + fn + '"']
+
+        self.setup_dyn_res(resname, restype, resval)
+
+        # Submit job
+        a = {'Resource_List.foo': '"red,blue"'}
+        j = Job(TEST_USER, attrs=a)
+        jid = self.server.submit(j)
+
+        # The job shouldn't run
+        job_comment = "Can Never Run: Insufficient amount of server resource:"
+        job_comment += " foo (red,blue != red blue)"
+        a = {'job_state': 'Q', 'comment': job_comment}
+        self.server.expect(JOB, a, id=jid, attrop=PTL_AND)
+
+    def test_res_string_array_invalid_input2(self):
+        """
+        Test invalid values returned from server_dyn_resource
+        script for resource type 'string_array'.
+        Script returns 'int' instead of type 'string_array'.
+        """
+        # Create a resource of type string_array
+        resname = ["foo"]
+        restype = ["string_array"]
+
+        # Prep for server_dyn_resource script
+        script_body = "echo 1,2,3"
+
+        fn = self.du.create_temp_file(prefix="PtlPbs_str_arr",
+                                      suffix=".scr",
+                                      body=script_body)
+        self.du.chmod(path=fn, mode=0755, sudo=True)
+
+        resval = ['"' + resname[0] + ' ' + '!' + fn + '"']
+
+        self.setup_dyn_res(resname, restype, resval)
+
+        # Submit job
+        a = {'Resource_List.foo': '"1,2,3"'}
```

Comment:
You are requesting a string "1,2,3", not three values of a string array.  If you were to request 1 or 2 or 3, your job would run.  The server_dyn_res script is not returning ints.  It's returning three values for the string array ("1", "2", and "3").